### PR TITLE
Updated nixpkgs-stable, nixd; added docker-compose module

### PR DESCRIPTION
**Pull request by airportyh**
Why
===

We need the latest version of docker compose.

What changed
============

1. updated nixpkgs-unstable
3. updated nixd flake as well because it broke after the nixpkgs-unstable update

Test plan
=========

1. create a repl
2. add "docker" to `modules` array in `.replit`
3. try to use docker-compose in shell and see its version is 2.27.0.

### CODE REVIEW

File: pkgs/modules/default.nix [hunk 1]
```diff
@@ -60,6 +60,7 @@ let
     (import ./dart)
     (import ./deno)
     (import ./docker)
+    (import ./docker-compose)
```
Comment (ryantm) [inline pkgs/modules/default.nix:63]: https://github.com/replit/nixmodules/blob/4d6f7cd9fd685a3319ac7b6e3fc0789b430d6289/pkgs/modules/docker/default.nix#L182 docker-compose is already part of the docker module. I think we don't need to add another module for this.

File: pkgs/historical-modules/default.nix [hunk 2]
```diff
@@ -43,6 +43,14 @@ let
       moduleId = "haskell-ghc9.2";
       commit = "4c6f2315da24b84bd5e9dfedb952e41677724aaa";
     }
+    {
+      moduleId = "haskell-ghc9.4";
```
Comment (airportyh) [inline pkgs/historical-modules/default.nix:47]: latest is haskell-ghc9.6

File: pkgs/historical-modules/default.nix [hunk 3]
```diff
@@ -43,6 +43,14 @@ let
...
+      commit = "4d6f7cd9fd685a3319ac7b6e3fc0789b430d6289";
+    }
+    {
+      moduleId = "elixir-1_15";
```
Comment (airportyh) [inline pkgs/historical-modules/default.nix:51]: latest is elixir-1_16

**Review by ryantm — APPROVED** [review]

**Comment by Juanito98** [comment]
🚢

File: pkgs/modules/python/pip.nix [hunk 4]
```diff
@@ -1,7 +1,16 @@
...
-  pip = pypkgs.pip;
+  pip = pypkgs.pip.overridePythonAttrs (old: rec {
+    outputs = [
+      "out"
+    ];
+    # Skip building the docs for pip because that was failing
+    # with a sphinx-build error with a version of nixpkgs-unstable for 3.10 (worked for >3.11)
+    # and we don't need the docs
+    postBuild = "";
+    postInstall = "";
```
Comment (ryantm) [inline pkgs/modules/python/pip.nix:12]: Are we also going to lose [shell completions here](https://github.com/NixOS/nixpkgs/blob/9ddcaffecdf098822d944d4147dd8da30b4e6843/pkgs/development/python-modules/pip/default.nix#L96)? It looks like they aren't working anyway already though.

**Review by ryantm — APPROVED** [review]

File: pkgs/modules/python/pip.nix (hunk 4 above)
Comment (airportyh) [reply pkgs/modules/python/pip.nix:12]: Oops I can add them back in another pr (ci will be quicker).

**Pull request merged**